### ISSUE TITLE
How to get AdMob Ads to render nicely when using native page transition animation as I navigate different screens

### ISSUE THREAD

**Issue by awongCM** [floatinghotpot/cordova-admob-pro#282]
I'm using this plugin animation from Telerik.

http://plugins.telerik.com/cordova/plugin/native-page-transitions

On my app, I have the home that some content and it has Admob ad displayed at the bottom of the screen, such as this

```
var newHomePageView = new HomePageView(),
      newSettingsPageView = new HomePageView();

//assuming Admob banner instance has been created at this point.
 
somerouter.addRoute('homepage', function() {
  $("body").html(newHomePageView.render());
  AdMob.showBanner( AdMob.AD_POSITION.BOTTOM_CENTER);
})

somerouter.addRoute('settingpage', function() {
  $("body").html(newSettingsPageView.render());
  AdMob.hideBanner();
});
```

What the above-mentioned code does is that I have single page mobile app which allows me to render two different content pages when navigating between home page and settings page.  I want the AdMob banner to only be displayed at the bottom of home page.  It should not be displayed on the settings page because Admob is a global object for the entire window DOM element as far as I understand its API design.  Hence I hide it.

Everything works as expected when navigating between the two pages.

Now when I tried to add in the page transition animation using the plugin(as per its guide) above;

```
somerouter.addRoute('homepage', function() {
  $("body").html(newHomePageView.render());
  AdMob.showBanner( AdMob.AD_POSITION.BOTTOM_CENTER);
  window.plugins.nativepagetransitions.fade(
      {"duration": right}, undefined, undefined
  );
})

somerouter.addRoute('settingspage', function() {
  $("body").html(newSettingsPageView.render());
  AdMob.hideBanner();
  window.plugins.nativepagetransitions.fade(
      {"duration": left}, undefined, undefined
  );
});
```

The transition behaviour works correctly for my content pages I generated using Jquery html methods, but it doesn't work for AdMob when hiding or showing banners.  I'm just wondering is it because of the plugin's limitation that only works with user generated content, but not for Admob?  Just because the transition plugin has no control over Admob content? 

I'll look forward to hearing your thoughts on this.

**Comment by floatinghotpot** [comment]
Yes, you are correct. the transition plugin has no control over Admob content.

The AdMob banner view is a native view, it's not a DOM inside the webview. 
You can only simply show/hide it with showBanner()/hideBanner().

**Comment by awongCM** [comment]
Hi @floatinghotpot,

Thanks for your prompt reply.

That's a shame.

So there's no other suitable mobile front-end framework that would allow me to do this besides Cordova?  Does it mean I can only achieve this behaviour by writing my own Admob natively from scratch instead of using your plugin?

**Comment by floatinghotpot** [comment]
I can try improving my plugin with smooth transition. Instead of writing it from scratch, can you just contribute some idea or reference code snippet?

**Comment by awongCM** [comment]
Hmm...  Good point.

Though I don't know much about Android ecosystem.

But I found this link to be interesting info for some ideas.

http://androidforbegineers.blogspot.com.au/2013/08/android-slide-animation-effect-between.html

It's only for pure native transitions.

But in order to implement this feature for your plugin,  my thoughts would involve having to provide extra arguments/callback functions inside showBanner/hideBanner method?

https://github.com/floatinghotpot/cordova-admob-pro/wiki/1.3-Methods-for-Banner

for eg

```
AdMob.showBanner(position, transitionType, duration, direction )
```
where 
`transitionType` can either slide or flip or fade type. I'd imagine these would have to some sort of const values in the native view.
`duration` is the duration of the transition takes place to complete animation.
`direction` is either left, right, up, down depending on the transitionType you set out in the beginning.

Don't know if the're any much help to you but this is how I would start thinking out loud about them.
